python/katana/local/analytics/_ksssp.pyx

```diff
@@ -0,0 +1,209 @@
+        """
+        if graph is None:
+            self.underlying_ = _KssspPlan()
+        else:
+            self.underlying_ = _KssspPlan(underlying_property_graph(graph))
+
+    Algorithm = _KssspAlgorithm
+    Reachability = _KssspReachability
+
+    @property
+    def algorithm(self) -> _KssspAlgorithm:
+        """
+        The selected algorithm.
+        """
+        return _KssspAlgorithm(self.underlying_.algorithm())
+
+    @property
+    def reachability(self) -> _KssspReachability:
+        """
+        The selected reachability method
+        """
+        return _KssspReachability(self.underlying_.reachability())
+
+    @property
+    def delta(self) -> int:
+        """
+        The exponent of the delta step size (2 based). A delta of 4 will produce a real delta step size of 16
+        """
+        return self.underlying_.delta()
+    @property
+    def edge_tile_size(self) -> int:
+        """
+        The edge tile size.
+        """
+        return self.underlying_.edge_tile_size()
+
+    def __init__(self):
+        """
+        Choose an algorithm using heuristics
+        """
+        super(KssspPlan, self).__init__()
+
+    @staticmethod
+    def delta_tile(_KssspPlan.Reachability reachability = kDefaultReach, unsigned delta = kDefaultDelta,
+                   ptrdiff_t edge_tile_size = kDefaultEdgeTileSize) -> KssspPlan:
+        """
+        Delta stepping tiled
+        """
+        return KssspPlan.make(_KssspPlan.DeltaTile(reachability, delta, edge_tile_size))
+
+    @staticmethod
+    def delta_step(_KssspPlan.Reachability reachability = kDefaultReach, unsigned delta = kDefaultDelta) -> KssspPlan:
+        """
+        Delta stepping (non-tiled)
+        """
+        return KssspPlan.make(_KssspPlan.DeltaStep(reachability, delta))
+
+    @staticmethod
+    def delta_step_barrier(_KssspPlan.Reachability reachability = kDefaultReach, unsigned delta = kDefaultDelta) -> KssspPlan:
+        """
+        Delta stepping with barrier
+        """
+        return KssspPlan.make(_KssspPlan.DeltaStepBarrier(reachability, delta))
+
+
+def ksssp(pg, str edge_weight_property_name, size_t start_node,
+          size_t report_node, size_t num_paths, bool is_symmetric=False,
+          KssspPlan plan = KssspPlan(), *, txn_ctx = None):
+    """
+    Compute the K-Shortest Path on `pg` using `start_node` as source.
+
+    :type pg: katana.local.Graph
+    :param pg: The graph to analyze
+    :type edge_weight_property_name: str
+    :param edge_weight_property_name: The input property containing edge weights.
+    :type start_node: Node ID
+    :param start_node: The source node
+    :type report_node: Node ID
+    :param report_node: The destination node
+    :type num_paths: int
+    :param num_paths: Number of paths to look for
+    :type is_symmetric: bool
+    :param is_symmetric: Whether or not the graph is symmetric. Defaults to false.
```

Comment:
This shouldn't be here. I can accept it as a short term fix until we support views, but it's bad.